src/dynamicprompts/samplers/utils.py

```diff
@@ -1,28 +1,27 @@
-from dynamicprompts.commands import (
-    VariantCommand,
-    WildcardCommand,
-)
+from dynamicprompts.commands import VariantCommand, VariantOption, WildcardCommand
+from dynamicprompts.parser.parse import parse
 from dynamicprompts.sampling_context import SamplingContext
 
 
 def wildcard_to_variant(
-    wildcard: WildcardCommand,
+    command: WildcardCommand,
     *,
     context: SamplingContext,
     min_bound=1,
     max_bound=1,
     separator=",",
 ) -> VariantCommand:
-    wm = context.wildcard_manager
-    wildcard_values = wm.get_all_values(wildcard.wildcard)
-    min_bound = min(min_bound, len(wildcard_values))
-    max_bound = min(max_bound, len(wildcard_values))
+    values = context.wildcard_manager.get_all_values(command.wildcard)
+    min_bound = min(min_bound, len(values))
+    max_bound = min(max_bound, len(values))
 
-    wildcard_variant = VariantCommand.from_literals_and_weights(
-        wildcard_values,
-        min_bound=min_bound,
-        max_bound=max_bound,
-        separator=separator,
-        sampling_method=wildcard.sampling_method,
+    variant_options = [VariantOption(parse(v)) for v in values]
```

Comment:
This won't take a custom parser config that could have been in use when originally parsing things into account. If we need that, we'll probably need to schlep that along in the context too.